server/src/main/java/org/opensearch/bootstrap/jvm/DenyJvmVersionsParser.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.bootstrap.jvm;
+
+import org.opensearch.common.Nullable;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.lang.Runtime.Version;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Parses the list of JVM versions which should be denied to run Opensearch engine with due to discovered
+ * issues or flaws.
+ */
+public class DenyJvmVersionsParser {
+    public interface VersionPredicate extends Predicate<Version> {
+        String getReason();
+    }
+
+    private static class SingleVersion implements VersionPredicate {
+        final private Version version;
+        final private String reason;
+
+        public SingleVersion(Version version, String reason) {
+            this.version = version;
+            this.reason = reason;
+        }
+
+        @Override
+        public boolean test(Version v) {
+            return version.compareTo(v) == 0;
+        }
+
+        @Override
+        public String getReason() {
+            return reason;
+        }
+    }
+
+    private static class VersionRange implements VersionPredicate {
```

Comment:
I have a feeling we'll use `VersionRange` for things like plugin compatibility loading in the future... For now it's ok in private, just thinking out loud.

Reply:
It would be different version though - Opensearch version, not Runtime.Version, but that's more a detail, thanks!